### PR TITLE
Added WrapperPlayClientCraftRecipeRequest & WrapperPlayServerCraftRecipeResponse created

### PR DESCRIPTION
**Pull request by NoJokeFNA**
With this pull request users can use two new wrappers (WrapperPlayClientCraftRecipeRequest & WrapperPlayServerCraftRecipeResponse) that work fine - I think there is nothing more to explain.
It was tested with the server versions: 1.8.8, 1.12.2, 1.13, 1.13.2, 1.14.4, 1.15.2, 1.16.5 and 1.18.2
Client-Versions: 1.18.2, 1.16.5, 1.13
Java-Versions: 8, 11, 16, 17

### CODE REVIEW

**Review by retrooper — CHANGES_REQUESTED** [review]
I'd say revert the debugs. Or atleast don't register the debug listener in the PacketEventsPlugin class. Otherwise looks fine to me.

**Comment by NoJokeFNA** [comment]
Done!

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/client/WrapperPlayClientCraftRecipeRequest.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+  public WrapperPlayClientCraftRecipeRequest(int windowId, Object recipe, boolean makeAll) {
+    super(PacketType.Play.Client.CRAFT_RECIPE_REQUEST);
+    this.windowId = windowId;
+    this.recipe = recipe;
+    this.makeAll = makeAll;
+  }
+
+  @Override
+  public void read() {
+    if (serverVersion.isNewerThanOrEquals(ServerVersion.V_1_12)) {
```
Comment (MWHunter) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/client/WrapperPlayClientCraftRecipeRequest.java:44]: Did you mean specifically 1.12?

**Review by MWHunter — CHANGES_REQUESTED** [review]

**Comment by retrooper** [comment]
I didn't look into this too deply, but I'm not a fan of Objects in wrappers. It's an API.

**Review by MWHunter — APPROVED** [review]

**Comment by retrooper** [comment]
Might merge this soon, but i wish it was possible to bring the recipe to one form without legacy and modern variables. I’ll see in a bit.

**Pull request merged**